app/dashboard/views.py

```diff
@@ -2687,21 +2703,32 @@ def get_profile_tab(request, profile, tab, prev_context):
         pass
     elif tab == 'portfolio':
         title = request.POST.get('project_title')
-        if title:
-            if request.POST.get('URL')[0:4] != "http":
-                messages.error(request, 'Invalid link.')
-            elif not request.POST.get('URL')[0:4]:
-                messages.error(request, 'Please enter some tags.')
-            elif not request.user.is_authenticated or request.user.profile.pk != profile.pk:
-                messages.error(request, 'Not Authorized')
-            else:
-                PortfolioItem.objects.create(
-                    profile=request.user.profile,
-                    title=title,
-                    link=request.POST.get('URL'),
-                    tags=request.POST.get('tags').split(','),
-                    )
-                messages.info(request, 'Portfolio Item added.')
+        portfolio_id = request.POST.get('id')
+        if portfolio_id:
+            portfolio_item = PortfolioItem.objects.get(id = portfolio_id)
+            portfolio_item.title = title
+            portfolio_item.profile = request.user.profile
+            portfolio_item.id = portfolio_id
+            portfolio_item.link = request.POST.get('URL')
+            portfolio_item.tags = request.POST.get('tags').split(',')
```

Comment:
if there are no tags, this will return `None` and then `None.split(',')` will throw an error. To overcome this, the 2nd argument to `.get` allows you to add your own value instead of None, so lets use an empty string.

```
Python 3.7.3 (default, Oct 11 2019, 19:39:43)
[Clang 11.0.0 (clang-1100.0.33.12)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> ''.split(',')
['']
>>> None.split(',')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'split'`